src/cloudwatch-applicationsignals-mcp-server/awslabs/cloudwatch_applicationsignals_mcp_server/enablement_guides/templates/eks/eks-java-enablement.md

```diff
@@ -0,0 +1,211 @@
+# Enable Application Signals for Java Applications on Amazon EKS
+
+This guide shows how to modify the existing CDK and Terraform infrastructure code to enable AWS Application Signals for Java applications running on Amazon EKS.
+
+## Prerequisites
+
+- Application Signals enabled in your AWS account (see [Enable Application Signals in your account](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Application-Signals-Enable.html))
+- Existing EKS cluster deployed using the provided CDK or Terraform code
+- Java application containerized and pushed to ECR
+- AWS CLI configured with appropriate permissions
+
+## Critical Requirements
+
+**Error Handling:**
+- If you cannot determine required values from the IaC, STOP and ask the user
+- For multiple EC2 instances, ask which one(s) to modify
+- Preserve all existing UserData commands; add new ones in sequence
+
+**Do NOT:**
+- Run deployment commands automatically (`cdk deploy`, `terraform apply`, etc.)
+- Remove existing application startup logic
+- Skip the user approval step before deployment
+
+## CDK Implementation
+
+### 1. Enable Application Signals Discovery (Optional)
```

Comment:
Does the (Optional) still add startDiscovery in IaC? If yes, we can remove it at this moment before the idempotent problem is fixed.